web/src/components/tasks/MultipleDwellingRegistrationTask.tsx

```diff
@@ -0,0 +1,172 @@
+import * as api from "@/lib/api-client/apiClient";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useRoadmap } from "@/lib/data-hooks/useRoadmap";
+import { MultipleDwellingSearchError, Task } from "@/lib/types/types";
+import { openInNewTab } from "@/lib/utils/helpers";
+import { getModifiedTaskContent } from "@/lib/utils/roadmap-helpers";
+import {
+  HousingAddress,
+  HousingRegistrationRequestLookupResponse,
+} from "@businessnjgovnavigator/shared/housing";
+import { TabContext, TabList, TabPanel } from "@mui/lab/";
+import { Box, Tab } from "@mui/material";
+import React, { ReactElement, useContext, useState } from "react";
+
+interface Props {
+  task: Task;
+  CMS_ONLY_disable_overlay?: boolean;
+}
+
+const APPLICATION_TAB_INDEX = 0;
+const STATUS_TAB_INDEX = 1;
+
+export const MultipleDwellingRegistrationTask = (props: Props): ReactElement => {
+  const { roadmap } = useRoadmap();
+  const callToActionLink = getModifiedTaskContent(roadmap, props.task, "callToActionLink");
+  const [tabIndex, setTabIndex] = useState(APPLICATION_TAB_INDEX);
+  const [error, setError] = useState<MultipleDwellingSearchError | undefined>(undefined);
+  const [housingAddress, setHousingAddress] = useState<HousingAddress | undefined>(undefined);
+  const [housingRegistrationLookupSummary, setHousingRegistrationLookupSummary] = useState<
+    HousingRegistrationRequestLookupResponse | undefined
+  >(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const { municipalities } = useContext(HousingMunicipalitiesContext);
+  const { Config } = useConfig();
+
+  const onSelectTab = (event: React.SyntheticEvent, newValue: string): void => {
+    const index = Number.parseInt(newValue);
+    setTabIndex(index);
+  };
+
+  const onEdit = (): void => {
+    setHousingAddress(undefined);
+    setHousingRegistrationLookupSummary(undefined);
+  };
+
+  if (municipalities.length === 0) {
+    setError(`SEARCH_FAILED`);
+  }
+
+  const onSubmit = (address: HousingAddress): void => {
+    if (address?.address1 && address?.municipalityExternalId) {
+      setIsLoading(true);
+      api
+        .checkHousingRegistrationStatus(address.address1, address.municipalityExternalId, "multipleDwelling")
+        .then((result) => {
+          if (result.lookupStatus === "NO PROPERTY INTERESTS FOUND") {
+            setError("NO_PROPERTY_INTEREST_FOUND");
+          } else if (result.lookupStatus === "NO REGISTRATIONS FOUND") {
+            setError("NO_MULTIPLE_DWELLINGS_REGISTRATIONS_FOUND");
+          } else {
+            setError(undefined);
+            setHousingAddress(address);
+            setHousingRegistrationLookupSummary(result);
+          }
+        })
+        .catch(() => {
+          setError("SEARCH_FAILED");
+        })
+        .finally(async () => {
+          setIsLoading(false);
+        });
+    } else {
+      setError("FIELDS_REQUIRED");
+      return;
```

Comment:
[dust] Maybe have this the first check and act like a guard clause? If you did that could you take away the first `if` and un-nest the logic in lines 61-80?

Reply:
Love it, I'll move that around